scripts/code_style.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+"""Check or fix the code style by running Uncrustify.
+"""
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+import io
+import os
+import subprocess
+import sys
+from typing import List
+
+CONFIG_FILE = "codestyle.cfg"
+UNCRUSTIFY_EXE = "uncrustify"
+UNCRUSTIFY_ARGS = ["-c", CONFIG_FILE]
+STDOUT_UTF8 = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8')
+STDERR_UTF8 = io.TextIOWrapper(sys.stderr.buffer, encoding='utf-8')
```

Comment:
Just for my own education: why do we need this? I though in modern python everything was UTF-8 by default already.

Reply:
The CI uses the 'ascii' encoding for some reason. Some of the files have non-ascii characters so when printing the diffs of files we need 'utf-8'.

Reply:
Ok, thanks!